### ISSUE TITLE
UniVL: A Unified Video and Language Pre-Training Model for Multimodal Understanding and Generation

### ISSUE THREAD

**Issue by yskim0** [yskim0/One-Day-One-Paper#57]
![image](https://user-images.githubusercontent.com/48315997/129238894-919cbc90-ef8c-4876-9960-eb752ba39767.png)

# Abstract

- Components (4)

> It comprises four components, including **two singlemodal encoders, a cross encoder, and a decoder with the Transformer backbone.**

- Objectives (5)

> video-text joint, conditioned masked language model (CMLM), conditioned masked frame model (CMFM), videotext alignment, and language reconstruction

- Two Stream Design

> stage by stage pre-training (StagedP) and enhanced video representation (EnhancedV)

- Pretrained Dataset : H100M
- Understanding task 뿐만 아니라 Generation task 또한 같이 pre-training 했다는 것이 이전 모델들(ActBERT, HERO)과의 차이점임.

# MODEL

1. input : Video Clip, Transcript(by ASR)
2. 각 input별 feature extractor를 사용
    - Text Encoder : **BERT** model to embed the text
    - Video Encoder : **Transformer** encoder to embed the video frames
3. [2] are fed to **Cross Encoder**
4. [3] are fed to **Transformer Decoder** to reconstruct the input text

## Method

> The problem is defined as: given the input video and the corresponding ASR transcript pairs, pretrain a model to learn joint video and text representation with the self-supervision approach, and fine-tune downstream tasks.

## Objectives Detail

1. Video-text joint objective
    - 두 인코더에 적용
    - 합치기 전에 각각의 modality에 대해 더 나은 표현을 학습하도록 함.
        - text encoder의 경우 robust한 extractor이나 Video encoder는 이에 비해 약한가 봄
        - video encoder의 capability를 향상시키기 위해 하는 작업임
2. **[2~5 : Transformer based encoder-decoder]** 
CMLM (conditioned masked language model)
    - for language corruption

    > Following BERT, we also randomly mask 15% tokens with the special token [MASK] in the sentence and re-produce the masked tokens under the condition of video input and known tokens.

3. CMFM (conditioned masked frame model)
    - for video corruption

    > we also propose a masked frame model to predict the correct frames given contextual frames and the input text for semantic constraints. However, it is hard to reconstruct the original RGB frame. We adopt the contrastive learning method to maximize the MI (Mutual information) between the masked output features and the original features.

    - frame을 masking 했을 때 이를 다시 복원(RGB로)하기는 어려우니 contrastive learning 방식을 사용한다.
4. Video-text alignment

    > We use the fused representation that corresponds to the special token [CLS] to predict scores for the video-text alignment, which is similar to the BERT sentence pair classification task.

5. Language reconstruction

## Pre-training Strategy

1. `StagedP`

    > We only pretrain the text encoder and video encoder by **the** **video-text joint objective** for the first stage.

2. `EnhancedV`

    > all modules will be pre-trained under **the whole objectives** in the second stage

    > Besides, we adopt **an entire masking strategy** EnhancedV on text to enhance video representation.

# Experiments

## Dataset

- ASR transcript from H100M
- YouCook2, MSR-VTT, COIN, CrossTask, CMU-MOSI

## Implementation

- Text Encoder
    - WordPiece Embeddings with a 30,000 token vocab.
- Video Encoder
    - 3D feature extractor : S3D model
    - Transformer Encoder

# Discussion

- ASR 사용한다는 점에서 일반적인 영상의 description이 잘된다는 보장이 없음.
    - *ActivityNet-Captions에 대한 결과가 궁금하다*
- inference time check하고 싶음
